### PR TITLE
MINIFICPP-2423 Fix RHEL8 bootstrap issues

### PR DESCRIPTION
**Pull request by szaszm**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -246,6 +246,16 @@ if [ -z "${CMAKE_COMMAND}" ] ||
   get_cmake_version
 fi
 
+# RHEL8: If cmake3 is too old, try cmake
+if [ "$CMAKE_COMMAND" = "cmake3" ] &&
+   [ "$CMAKE_MAJOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MAJOR" ] ||
+   [ "$CMAKE_MINOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MINOR" ] ||
+   [ "$CMAKE_REVISION" -lt "$CMAKE_GLOBAL_MIN_VERSION_REVISION" ] &&
```
Comment (lordgamez) [inline bootstrap.sh:253]: I think there are 2 problems with this:
- Due to the operator precedence the operands are evaluated in this order `([ "$CMAKE_COMMAND" = "cmake3" ] && [ "$CMAKE_MAJOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MAJOR" ]) || [ "$CMAKE_MINOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MINOR" ] || ([ "$CMAKE_REVISION" -lt "$CMAKE_GLOBAL_MIN_VERSION_REVISION" ] && [ -x "$(command -v cmake)" ])` while all the parts of the cmake version should be checked together
- I think the cmake version check doesn't work as it supposed to not just here but on line 235 and line 259 as well. For example with this check if the required version is 3.24.1 and our version is 3.25.0 it considers it lower than the required.

Comment (szaszm) [reply bootstrap.sh]: both should be fixed now

File: bstrp_functions.sh [hunk 2]
```diff
@@ -26,6 +26,24 @@ get_cmake_version(){
   CMAKE_REVISION=$(echo "$CMAKE_VERSION" | cut -d. -f3)
 }
 
+# Checks if a version is smaller than another
+compare_version() {
```
Comment (fgerlits) [inline bstrp_functions.sh:30]: I think something like `version_is_less_than` would be better than `compare_version`

File: bstrp_functions.sh [hunk 3]
```diff
@@ -457,3 +475,5 @@ read_feature_options(){
     *) echo -e "${RED}Please enter an option A-Z or AA-AF or 1-7...${NO_COLOR}" && sleep 2
   esac
 }
+
+# vim: shiftwidth=2 tabstop=2 expandtab
```
Comment (fgerlits) [inline bstrp_functions.sh:479]: did you add these vim-specific comments intentionally?

Comment (szaszm) [reply bstrp_functions.sh:479]: yes

File: bstrp_functions.sh (hunk 2 above)
Comment (szaszm) [reply bstrp_functions.sh]: renamed

**Review by fgerlits — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**